Lib/cgitb.py

```diff
@@ -85,7 +85,10 @@ def scanvars(reader, frame, locals):
         if ttype == tokenize.NAME and token not in keyword.kwlist:
             if lasttoken == '.':
                 if parent is not __UNDEF__:
-                    value = getattr(parent, token, __UNDEF__)
+                    try:
+                        value = getattr(parent, token, __UNDEF__)
+                    except Exception:
```

Comment:
Don't catch all subclasses of Exception. Make it more specific.

Reply:
Actually this getattr should not raise exceptions at all because it is given a default value. 

The docs for __getattr__ say: "This method should either return the (computed) attribute value or raise an AttributeError exception."    It is an error for it to raise a different exception, so I think this patch should not be merged.

Reply:
good catch.

Reply:
Well, the issue is saying this is error handling and the code should be more robust in the presence of such malformed objects, so maybe this patch can be merged, but in any case you should not swallow SystemError/MemoryError/OverflowErrors - they should propagate to the user.

Reply:
So I found an existing better PR for this:  #15094 related to https://bugs.python.org/issue966992

I will close this PR here. Thanks @iritkatriel for the pushback, it helped me dig into the issue a bit more.